### DISCUSSION TITLE
Display all relations from/to the software system on container views

### DISCUSSION THREAD

**Discussion by emredaglikfz** [structurizr/java#311] [Q&A]
We want to display all relations from/to the software system `s2` on the container view of `s2`. But on this view, we can only see the relations from/to containers of `s2`. 

Is it possible to show all relations from/to the software system `s2` on container views?

Thanks for your help 🙏

```
workspace {
    model {
        user = person "User"

        s1 = softwareSystem "s1" {
            c1 = container "c1"
        }

        s2 = softwareSystem "s2" {
            c2 = container "c2"
        }

        s1 -> s2
        // Or:
        c1 -> s2
        // Or:
        user -> s2
    }

    views {
        container s2 {
            include *
            autoLayout
        }
    }
}
```

The result is:

<img width="564" alt="image" src="https://github.com/user-attachments/assets/0e59b803-c4e8-4d99-8d5b-a61578bf3881">

**Comment by simonbrowndotje** [comment]
A container view shows relationships to the containers inside the scoped software system only I'm afraid ... there's no way to show relationships to/from the `s2` box.

**Comment by quentinproust** [comment]
@simonbrowndotje Do you have advices about this ?

In a more slightly more complex example, I can have the following model : 

```
workspace {
    !identifiers hierarchical
    model {
        user = person "User"

        s1 = softwareSystem "s1" {
            c1 = container "c1"
        }

        s2 = softwareSystem "s2" {
            public_api = container "api"
            c2 = container "c2"
            db = container "db"
        }

        user -> s1
        s1.c1 -> s2.db
        s1.c1 -> s2.c2
    }

    views {
        container s1 {
            include *
            autoLayout
        }
        container s2 {
            include *
            autoLayout
        }
    }
}
```

The rendered views : 

![image](https://github.com/user-attachments/assets/d42baa48-f625-4748-a796-f9d0d035bdb5)

![image](https://github.com/user-attachments/assets/a1366454-3b33-48d8-80ac-55dde26453b7)

In both rendered views, I can see that systems `s1` and `s2` are connected to each other but I have no idea how. 

In this example, I would prefer that `s1` use `s2` public api instead of internal containers like a direct connection to the db. But this information in not visible in the views si I might miss this point during an architecture review for example.

How do you handle these use case ? 

Note that I like the current view as it allow to a quick understanding of the system used but sometimes would like to have more details 😄

**Comment by quentinproust** [comment]
I just stumble on this issue https://github.com/structurizr/java/issues/358 that is talking about something similar.

By addind this view, I can se all relationships from s1 containers, it works great

```
container s1 "x" {
    include element.parent==s1
    include "element.parent==s2 && element==s1.c1->"
    autolayout lr
}
```

And I see that is documented here : https://docs.structurizr.com/dsl/cookbook/container-view-multiple-software-systems/

I just had to search a little more 😄

**Reply by chipbite** [reply]
Good thread. Maybe mark this as an answer?
